lib/test_cleanup.rb

```diff
@@ -31,9 +31,13 @@ class TestCleanup < Test
     ]).freeze
 
     def reset_if_needed(repository)
-      return if system(git, "-C", repository, "diff", "--quiet", "origin/master")
+      default_ref = Utils.safe_popen_read(
+        git, "-C", repository, "symbolic-ref", "refs/remotes/origin/HEAD", "--short"
+      ).strip
```

Comment:
This should be extracted into a function, use `popen_read`, appended with `.presence` (which will make it return `nil` if not `present?`) and be `||=`d to `origin/master` if it's `nil` (as I've found some local repositories where this will fail).